src/Data/Fin/Base.agda

```diff
@@ -234,6 +236,13 @@ _ℕ-ℕ_ : (n : ℕ) → Fin (suc n) → ℕ
 n     ℕ-ℕ zero   = n
 suc n ℕ-ℕ suc i  = n ℕ-ℕ i
 
+-- n ℕ-suc "i" = n ∸ (suc i)
+
+infixl 6 _ℕ-suc_
+
+_ℕ-suc_ : (n : ℕ) → Fin n → ℕ
+suc n ℕ-suc i = n ℕ-ℕ i
```

Comment:
Matthew, 
this function is not currently used, but since at various points in the library there are uses of its definitional equivalent `n ∸ suc (toℕ i)` I thought it worth lifting out as something in its own right; but admittedly in obscure corners:
* `Data.Fin.Properties.opposite-prop`, `Data.Fin.Properties.opposite-suc`
* `Data.List.Properties.lookup-applyDownFrom`, `Data.List.Properties.lookup-downFrom`

So very easy to prune this, and its properties, back. But it *is* an example of a 'naturally'-occurring homogenously-telescoped function on `Fin n`, so in the spirit of this PR.

Reply:
Okay, in which case could you make use of it in this PR to simplify the code that you mention? 

My deepest apologies if this was previously in this PR before and you have to add it back in. I'm just playing the part of Goldilocks. Previously the PR was too big to review, but now it contains too small in that it contains this partial change that doesn't stand on its own. In general, we tend to like each PR to stand on its own as people (including me!) promise to do things, but don't actually come back and do it! Not implying that you will, but as a general rule these things happen! Therefore we like each PR to be an obvious standalone improvement over the existing code. Adding a function that is never used doesn't quite pass this test.

Reply:
Don't mind, happy either way. Whichever you'd prefer!